src/Evolution/Systems/GrMhd/ValenciaDivClean/Sources.hpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+
+class DataVector;
+/// \endcond
+
+namespace grmhd {
+namespace ValenciaDivClean {
+
+/*!
+ * \brief Compute the source terms for the flux-conservative Valencia
+ * formulation of GRMHD with divergence cleaning.
+ *
+ *
+ * A flux-conservative system has the generic form:
+ * \f[
+ * \partial_t U_i + \partial_m F^m(U_i) = S(U_i)
+ * \f]
+ *
+ * where \f$F^a()\f$ denotes the flux of a conserved variable \f$U_i\f$ and
+ * \f$S()\f$ denotes the source term for the conserved variable.
+ *
+ * For the Valencia formulation:
+ * \f{align*}
+ * S({\tilde D}) = & 0\\
+ * S({\tilde S}_i) = & \frac{1}{2} \alpha {\tilde S}^{mn} \partial_i \gamma_{mn}
+ * + {\tilde S}_m \partial_i \beta^m - ({\tilde D} + {\tilde \tau}) \partial_i
+ * \alpha \\
+ * S({\tilde \tau}) = & \alpha {\tilde S}^{mn} K_{mn} - {\tilde S}^m \partial_m
+ * \alpha \\
+ * S({\tilde B}^i) = & -{\tilde B}^m \partial_m \beta^i + {\tilde \Phi}
+ * \gamma^{im} \partial_m \alpha + \alpha {\tilde \Phi} \left( \frac{1}{2}
+ * \gamma^{il} \gamma^{jk} - \gamma^{ij} \gamma^{lk} \right) \partial_l
+ * \gamma_{jk} \\ S({\tilde \Phi}) = & {\tilde B}^k \partial_k \alpha - \alpha K
+ * {\tilde \Phi}
+ * - \alpha \kappa {\tilde \Phi}
+ * \f}
+ *
+ * where
+ * \f{align*}
+ * {\tilde S}^i = & {\tilde S}_m \gamma^{im} \\
+ * {\tilde S}^{ij} = & \sqrt{\gamma} \left[ \left(h \rho W^2 + B^m B_m \right)
+ * v^i v^j + \left(p + p_m \right) \gamma^{ij} - B^m v_m \left( B^i v^j + B^j
```

Comment:
Could you avoid `m` as an index on the tensors since there's also `p_m` that is the magnetic pressure?

Reply:
A possible alternative: name the magnetic pressure `p_{\text{mag}}`